pkg/cmd/roachprod-microbench/main.go

```diff
@@ -162,16 +162,25 @@ func makeCompareCommand() *cobra.Command {
 
 		comparisonResult := c.createComparisons(metricMaps, "baseline", "experiment")
 
-		var links map[string]string
-		if config.publishGoogleSheet {
-			links, err = c.publishToGoogleSheets(comparisonResult)
-			if err != nil {
-				return err
+		if c.sheetDesc != "" {
+			var links map[string]string
+			if config.sheetDesc != "" {
+				links, err = c.publishToGoogleSheets(comparisonResult)
+				if err != nil {
+					return err
+				}
+			}
+
+			if config.slackConfig.token != "" {
```

Comment:
Good point, everything below `newCompare` should be using the `c` instance. Although it ends up referencing the same value, it should be updated to use the instance instead of the config directly. And yes, we should skip Slack if no sheets were generated. Updated.